processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -166,12 +169,66 @@ public String getType()
     return analysisTypes;
   }
 
+  public void setAnalysisTypes(EnumSet<AnalysisType> analysisTypes)
```

Comment:
Query object should be immutable. Please replace this method with "withAnalysisTypes()", returning a new query object with the given analysis types. Also update SegmentMetadataQueryBuilder, to include a new field.